mesh/mesh.go

```diff
@@ -317,6 +317,14 @@ func (vl *validator) ProcessLayer(ctx context.Context, layerID types.LayerID) er
 			return err
 		}
 	}
+	if !reverted && newVerified.Before(vl.latestLayerInState) {
```

Comment:
```suggestion
	} else if newVerified.Before(vl.latestLayerInState) {
```
looks a bit less redundant, but up to you

Reply:
i changed to a more robust variant